bot/cogs/extensions.py

```diff
@@ -0,0 +1,216 @@
+import functools
+import logging
+import typing as t
+from enum import Enum
+from pkgutil import iter_modules
+
+from discord import Colour, Embed
+from discord.ext import commands
+from discord.ext.commands import Bot, Context, group
+
+from bot.constants import Emojis, MODERATION_ROLES, Roles, URLs
+from bot.pagination import LinePaginator
+from bot.utils.checks import with_role_check
+
+log = logging.getLogger(__name__)
+
+UNLOAD_BLACKLIST = {"bot.cogs.extensions", "bot.cogs.modlog"}
+EXTENSIONS = frozenset(
+    ext.name
+    for ext in iter_modules(("bot/cogs",), "bot.cogs.")
+    if ext.name[-1] != "_"
+)
+
+
+class Action(Enum):
+    """Represents an action to perform on an extension."""
+
+    # Need to be partial otherwise they are considered to be function definitions.
+    LOAD = functools.partial(Bot.load_extension)
+    UNLOAD = functools.partial(Bot.unload_extension)
+    RELOAD = functools.partial(Bot.reload_extension)
+
+
+class Extension(commands.Converter):
+    """
+    Fully qualify the name of an extension and ensure it exists.
+
+    The * and ** values bypass this when used with the reload command.
+    """
+
+    async def convert(self, ctx: Context, argument: str) -> str:
+        """Fully qualify the name of an extension and ensure it exists."""
+        # Special values to reload all extensions
+        if ctx.command.name == "reload" and (argument == "*" or argument == "**"):
+            return argument
+
+        argument = argument.lower()
+
+        if "." not in argument:
+            argument = f"bot.cogs.{argument}"
+
+        if argument in EXTENSIONS:
+            return argument
+        else:
+            raise commands.BadArgument(f":x: Could not find the extension `{argument}`.")
+
+
+class Extensions(commands.Cog):
+    """Extension management commands."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @group(name="extensions", aliases=("ext", "exts", "c", "cogs"), invoke_without_command=True)
+    async def extensions_group(self, ctx: Context) -> None:
+        """Load, unload, reload, and list loaded extensions."""
+        await ctx.invoke(self.bot.get_command("help"), "extensions")
+
+    @extensions_group.command(name="load", aliases=("l",))
+    async def load_command(self, ctx: Context, extension: Extension) -> None:
```

Comment:
Is there any particular reason why you allow multiple extensions to be reloaded at once, but don't allow multiple extensions to be loaded or unloaded at once? This feels like an unnecessary break of symmetry between the sub-commands to me. 

Now, in your current version, we could use `reload` to `load` multiple extensions at once if they're unloaded, but that feels like a bit of a hack to me.

Reply:
It was like that in the old version so I left it. I could make them all support wildcards but I'd have to rewrite the batch function a bit to work to accept load and unload too. I don't mind looking into that.

Reply:
That's on me, then. I never noticed that we had this functionality in the old version, I thought you added this handy feature. I don't think it's necessary, it just struck me as an asymmetry when reading the code. I think it could be a nice refinement to make the sub-commands work in a consistent manner, but I'm not opposed to keeping the current version either. 

I'll leave this conversation open, but if no one else wants this, we can just close it. If we miss it down the line, we can always opt for a refinement later.

Reply:
I've add support for this now. The only caveat is that for `!ext load`, when multiple extensions are given, if any of those are already loaded, they will still be displayed as loaded (i.e. included in the total load count and no error shown). When a single extension is given, it will show an error if it's already loaded. The same applies to `!ext unload`. I figured this isn't a big deal and makes implementation simpler.

Reply:
That's totally fine with me.